src/bin/cargo-fmt.rs

```diff
@@ -97,7 +97,16 @@ fn execute() -> i32 {
     }
 }
 
-fn print_usage(opts: &Options, reason: &str) {
+fn print_usage_to_stderr(opts: &Options, reason: &str) {
+    let msg = format!("{}\nusage: cargo fmt [options]", reason);
```

Comment:
Could you please avoid duplicate here? The only difference between `print_usage_to_stderr` and `print_usage_to_stdout` is `println!` and `eprintln!`, so I think we can do better here.

Reply:
Do you have a preferred means to do this? I figured while the duplication is gross, it's minimal and I'd rather that than complicate the signature: a bool is hard to tell what it does at call sites, not aware of an appropriate enum (could add one just for this). Could have these functions shim through to a print_usage which takes a param to select output.

Reply:
I was thinking of using a macro. Something like the following will do:
```rust
macro_rules! print_usage {
    ($print:ident, $opts:expr, $reason:expr) => {
        let msg = format!("{}\nusage: cargo fmt [options]", $reason);
        $print!(
            "{}\nThis utility formats all bin and lib files of the current crate using rustfmt. \
             Arguments after `--` are passed to rustfmt.",
            opts.usage(&msg)
        );
    }
}

fn print_usage_to_stdout() { print_usage!(println, ...) }
fn print_usage_to_stderr() { print_usage!(eprintln, ...) }
```
I prefer not to duplicate the message string, as we may fail to keep them in sync at some point.

Reply:
Cheers for the explanation. Updated.